src/main/java/core/db/UserConfigurationTable.java

```diff
@@ -171,7 +172,7 @@ int getDBVersion() {
 	 * @param obj
 	 */
 	void store(Configuration obj) {
-		final HashMap<String, String> values = obj.getValues();
+		final Map<String, String> values = obj.getValues();
```

Comment:
just for curiosity is there any reason for this ?   usually I do the opposite I use hashmap instead of map, but honest I don't really what I am doing when doing this. I just thought it was best to be more specific ....

Reply:
It's a good question!  It is usually preferrable to code to abstraction (google “code to an interface, not an implementation” for research): this means that the client of the method only knows about the behaviour of the object, not its internal implementation.

For example, here `HashMap` is a specific implementation of a `Map`, but for some reason we could decide we'd prefer a `Hashtable`, `WeakHashMap`, a `TreeMap` or a `LinkedHashMap`: if we code to implementation, we need to change the type everywhere it's used; if we code to abstraction, you only change in one place. (an example of that is the recent `ImageIcon` change which required to change to `Icon` in lots of places)

Reply:
thanks for the detailled explanation

Reply:
With pleasure: you're usually the one taking the time to explain to me. 🙂